/Users/omidvart/Library/Python/3.9/lib/python/site-packages/stable_baselines3/common/vec_env/patch_gym.py:49: UserWarning: You provided an OpenAI Gym environment. We strongly recommend transitioning to Gymnasium environments. Stable-Baselines3 is automatically wrapping your environments in a compatibility layer, which could potentially cause issues.
  warnings.warn(
Traceback (most recent call last):
  File "/Users/omidvart/Desktop/Amazon/RL-Modular/train_policy.py", line 19, in <module>
    shortest_path_agent.learn(timesteps=10000)
  File "/Users/omidvart/Desktop/Amazon/RL-Modular/agent.py", line 22, in learn
    self.model.learn(total_timesteps=timesteps, callback=callback)
  File "/Users/omidvart/Library/Python/3.9/lib/python/site-packages/stable_baselines3/dqn/dqn.py", line 267, in learn
    return super().learn(
  File "/Users/omidvart/Library/Python/3.9/lib/python/site-packages/stable_baselines3/common/off_policy_algorithm.py", line 312, in learn
    rollout = self.collect_rollouts(
  File "/Users/omidvart/Library/Python/3.9/lib/python/site-packages/stable_baselines3/common/off_policy_algorithm.py", line 552, in collect_rollouts
    if callback.on_step() is False:
  File "/Users/omidvart/Library/Python/3.9/lib/python/site-packages/stable_baselines3/common/callbacks.py", line 104, in on_step
    return self._on_step()
  File "/Users/omidvart/Desktop/Amazon/RL-Modular/agent.py", line 6, in _on_step
    reward = self.training_env.get_attr('reward')[0]
  File "/Users/omidvart/Library/Python/3.9/lib/python/site-packages/stable_baselines3/common/vec_env/dummy_vec_env.py", line 116, in get_attr
    return [getattr(env_i, attr_name) for env_i in target_envs]
  File "/Users/omidvart/Library/Python/3.9/lib/python/site-packages/stable_baselines3/common/vec_env/dummy_vec_env.py", line 116, in <listcomp>
    return [getattr(env_i, attr_name) for env_i in target_envs]
  File "/Users/omidvart/Library/Python/3.9/lib/python/site-packages/gymnasium/core.py", line 282, in __getattr__
    return getattr(self.env, name)
  File "/Users/omidvart/Library/Python/3.9/lib/python/site-packages/shimmy/openai_gym_compatibility.py", line 218, in __getattr__
    return getattr(self.gym_env, item)
AttributeError: 'GridWorld' object has no attribute 'reward'
Using cpu device
Wrapping the env with a `Monitor` wrapper
Wrapping the env in a DummyVecEnv.